pippo-core/src/main/java/ro/fortsoft/pippo/core/Response.java

```diff
@@ -576,7 +576,7 @@ public void send(CharSequence content) {
     public void resource(InputStream input) {
         checkCommitted();
 
-        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE)) {
+        if (isHeaderEmpty(HttpConstants.Header.CONTENT_TYPE) && getHttpServletResponse().getContentType() == null) {
             header(HttpConstants.Header.CONTENT_TYPE, HttpConstants.ContentType.APPLICATION_OCTET_STREAM);
         }
 
```

Comment:
The documentation says:
>     /**
     * Gets the value of the response header with the given name.
     * 
     * <p>If a response header with the given name exists and contains
     * multiple values, the value that was added first will be returned.
     *
     * <p>This method considers only response headers set or added via
     * {@link #setHeader}, {@link #addHeader}, {@link #setDateHeader},
     * {@link #addDateHeader}, {@link #setIntHeader}, or
     * {@link #addIntHeader}, respectively.
     *
     * @param name the name of the response header whose value to return
     *
     * @return the value of the response header with the given name,
     * or <tt>null</tt> if no header with the given name has been set
     * on this response
     *
     * @since Servlet 3.0
     */
    public String getHeader(String name); 

and
>     /**
     * Returns the content type used for the MIME body
     * sent in this response. The content type proper must
     * have been specified using {@link #setContentType}
     * before the response is committed. If no content type
     * has been specified, this method returns null.
     * If a content type has been specified, and a
     * character encoding has been explicitly or implicitly
     * specified as described in {@link #getCharacterEncoding}
     * or {@link #getWriter} has been called,
     * the charset parameter is included in the string returned.
     * If no character encoding has been specified, the
     * charset parameter is omitted.
     *
     * @return		a <code>String</code> specifying the
     *			content type, for example,
     *			<code>text/html; charset=UTF-8</code>,
     *			or null
     *
     * @since Servlet 2.4
     */  
    public String getContentType();

So we can use our `Response.getContentType() in all our checks - internally this method calls `httpServletResponse.getContentType()`

We should use the same strategy for contentLenght